### DISCUSSION TITLE
Users Roles

### DISCUSSION THREAD

**Discussion by m-bakoulis** [givanz/Vvveb#221] [Q&A] [answered]
Hi, I am trying to create a user that will have some limited roles and I am not able to allow the user to change the status of the site from Live to Maintenance. When I am testing this user I am getting the:

> Your role does not have permission to access this action!
![Screenshot 2024-11-12 152418](https://github.com/user-attachments/assets/83647c9a-12bb-4696-a193-c0ac3b22319c)

How I can add those permissions on a user role?

Thanks in advance.

**Comment by givanz** [comment, marked as answer]
Hi

The permission rule used for site state is the same as site save `settings/site/save` 

I added precedence for exact rules in the latest update, you need to redowload `latest.zip` and update `vvveb/system/user/role.php`.

With the new change even if you have `*/save` or `settings/site/*`  in your deny list, adding an exact rule like `settings/site/save` in allow list will take precedence and give permission to the action.

You will need to logout and login again the new user after it's user role is updated to load the new permissions.

**Reply by m-bakoulis** [reply]
Thank you givaz for your quick response and for fixing this.
I just tried it and works normally.

**Reply by m-bakoulis** [reply]
Now I have another question.
Which permission is needed in order to give access to the messages screen?

**Reply by givanz** [reply]
For messages list is `plugins/contact-form/messages` and for message view page is `plugins/contact-form/message` 

The permission rule is the module path for the page, you can view it in the page address.

If the url also includes action such as `&action=save` then you also need to append `/save` like `plugins/contact-form/messages/save`

![contact-form-message](https://github.com/user-attachments/assets/5f725a51-e9a0-490b-8d50-a687fafa5dbe)
![contact-form-messages](https://github.com/user-attachments/assets/61539e33-cc3d-4846-bccb-cfb75ee4be9b)

**Reply by m-bakoulis** [reply]
I just tried it, but I don't see the Messages Entry under the Users Entry in the left side menu

**Reply by givanz** [reply]
In the last update I changed the code and made `index` action optional to make things easier, changed files `admin/controller/base.php` and `system/user/role.php`

Previously with the older code the rule had to include `/index` like `plugins/contact-form/messages/index`  and  `plugins/contact-form/message/index`

**Reply by m-bakoulis** [reply]
yes I tried it that way and now it works.
Thanks
